dotnet/packages/Microsoft.Bot.Builder.M365/Microsoft.Bot.Builder.M365/Application.cs

```diff
@@ -159,19 +172,22 @@ public async Task OnTurnAsync(ITurnContext turnContext, CancellationToken cancel
                 TState turnState = (TState)new TurnState();
 
                 // Call before activity handler
-                if (!await OnBeforeTurnAsync(turnContext, turnState, cancellationToken)) return;
+                if (!await this.OnBeforeTurnAsync(turnContext, turnState, cancellationToken))
+                {
+                    return;
+                }
 
                 // Call activity type specific handler
-                bool eventHandlerCalled = await RunActivityHandlerAsync(turnContext, turnState, cancellationToken);
+                bool eventHandlerCalled = await this.RunActivityHandlerAsync(turnContext, turnState, cancellationToken);
 
-                if (!eventHandlerCalled && _ai != null && ActivityTypes.Message.Equals(turnContext.Activity.Type, StringComparison.OrdinalIgnoreCase) && turnContext.Activity.Text != null)
+                if (!eventHandlerCalled && this._ai != null && ActivityTypes.Message.Equals(turnContext.Activity.Type, StringComparison.OrdinalIgnoreCase) && turnContext.Activity.Text != null)
                 {
                     // Begin a new chain of AI calls
-                    await _ai.ChainAsync(turnContext, turnState); 
+                    _ = await this._ai.ChainAsync(turnContext, turnState);
```

Comment:
Is there a reason we want to assign these calls to `_`

Reply:
There's warning due to [*IDE0058 (Remove unnecessary expression value)*](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/style-rules/ide0058) . I've removed that rule to keep consistent with existing code - we do need expression calls in code.

Reply:
Added IDE0058 back since it helps detect unnecessary calls.